presto-main/src/main/java/com/facebook/presto/sql/planner/EqualityInference.java

```diff
@@ -170,14 +169,14 @@ private Expression rewriteExpression(Expression expression, Predicate<Symbol> sy
      */
     public EqualityPartition generateEqualitiesPartitionedBy(Predicate<Symbol> symbolScope)
     {
-        Set<Expression> scopeEqualities = new HashSet<>();
-        Set<Expression> scopeComplementEqualities = new HashSet<>();
-        Set<Expression> scopeStraddlingEqualities = new HashSet<>();
+        ImmutableSet.Builder<Expression> scopeEqualities = ImmutableSet.builder();
```

Comment:
can you add a test for that?

Reply:
This is hard to test. I found that order is preserved across repetitive test execution, but might be changed when JVM is restarted. 

General practice should be to avoid `HashSet` and `HashMap` when we are going to iterate over it in planner.